### PR TITLE
added possibility for specifiying a path relative to the filesystem a…

### PR DESCRIPTION
**Pull request by HaaseIT**
…dapter

I am using flysystem for various file operations. With this, you can define a base filesystem and load the twig templates from a subdirectory.

I'm sure, there are more elegant ways to do this, feel free to criticise or enhance my edit.

### CODE REVIEW

File: src/FlysystemLoader.php [hunk 1]
```diff
@@ -17,16 +17,17 @@ class FlysystemLoader implements Twig_LoaderInterface
     /**
      * @var Filesystem
      */
-    private $filesystem;
+    private $filesystem, $templatepath;
```
Comment (cedricziel) [inline src/FlysystemLoader.php]: one initialization per line please, lowerCamelCase

File: src/FlysystemLoader.php [hunk 2]
```diff
@@ -42,7 +43,7 @@ public function getSource($name)
     {
         $this->getFileOrFail($name);
 
-        return $this->filesystem->read($name);
+        return $this->filesystem->read($this->templatepath.$name);
```
Comment (cedricziel) [inline src/FlysystemLoader.php]: I feel like doing the concatenation three times in a row is a candiate for extracting a method. What do you think?

**Comment by cedricziel** [comment]
Hi Marcus!

Thanks a bunch for your contribution!

I'd be happy to merge once travis is happy, please see https://travis-ci.org/cedricziel/twig-loader-flysystem/jobs/155084874#L313 - it refers to declaring two properties on the same line as I commented.

Also: I know the loader is pretty small - but I feel like we could concatenate the string once instead of three times, as I commented - what do you think?

**Comment by HaaseIT** [comment]
Moin Cedric,
I changed the declaration of the properties.

Feel free to change the concentration, this was only a quick hack for me, I'm deep in my project (hcsfng-frontend) right now, just wanted to make it work for me. With the next push I will include my fork / your merged package.

Viele Grüße

Marcus

**Comment by cedricziel** [comment]
Thx a bunch for the amendments!

**Comment by HaaseIT** [comment]
My pleasure! Thanks for supplying this package!

**Comment by cedricziel** [comment]
FYI: I just tagged v1.1.0 and it's live on packagist

**Comment by HaaseIT** [comment]
Thanks a lot.
Btw. I forgot to add the annotation for the new argument in the constructor. :/

**Comment by cedricziel** [comment]
Fixed :)

**Pull request merged**